src/events/stickyMessageSet.ts

```diff
@@ -0,0 +1,78 @@
+import { ChannelType, Events, TextChannel } from 'discord.js'
+
+import { prisma } from '@/prisma.js'
+
+import { STICKY_CACHE_PREFIX } from '../features/stickyMessage/index.js'
+import { defineEventHandler } from '../types/defineEventHandler.js'
+import { saveCache } from '../utils/cache.js'
+
+export default defineEventHandler({
+  eventName: Events.MessageCreate,
+  once: false,
+  execute: async (botContext, message) => {
+    if (!message.content.startsWith('?stickao-set')) {
+      return
+    }
```

Comment:
@samithiwat 

right now does it checks that the user setting/removing a sticky is actually allowed to do it? e.g. check for a certain role or permission.

one challenge of using a message handler is that it’s not a slash command, so there is no built-in access control system provided by Discord.